src/main/java/de/tum/in/www1/artemis/service/plagiarism/JPlagSubmissionDataExtractor.java

```diff
@@ -0,0 +1,96 @@
+package de.tum.in.www1.artemis.service.plagiarism;
+
+import static java.util.Comparator.naturalOrder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import de.jplag.Submission;
+import de.tum.in.www1.artemis.domain.DomainObject;
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.plagiarism.PlagiarismSubmission;
+import de.tum.in.www1.artemis.domain.plagiarism.text.TextSubmissionElement;
+import de.tum.in.www1.artemis.repository.SubmissionRepository;
+
+@Component
+class JPlagSubmissionDataExtractor {
+
+    private static final Logger log = LoggerFactory.getLogger(JPlagSubmissionDataExtractor.class);
+
+    private static final long DEFAULT_SUBMISSION_ID = 0;
+
+    private static final String DEFAULT_STUDENT_LOGIN = "unknown";
+
+    private final SubmissionRepository submissionRepository;
+
+    JPlagSubmissionDataExtractor(SubmissionRepository submissionRepository) {
+        this.submissionRepository = submissionRepository;
+    }
+
+    void getAndSetSubmissionIdAndStudentLogin(PlagiarismSubmission<TextSubmissionElement> submission, Submission jplagSubmission, Exercise exercise) {
```

Comment:
```suggestion
    void retrieveAndSetSubmissionIdAndStudentLogin(PlagiarismSubmission<TextSubmissionElement> submission, Submission jplagSubmission, Exercise exercise) {
```
This would make Codacy more happy (same for other methods)

Reply:
Done